### ISSUE TITLE
Does not detect nullish coalescing or optional chaining as lint error for defaults preset

### ISSUE THREAD

**Issue by David-Else** [amilajack/eslint-plugin-compat#360]
EDIT: I think this issues might cover it, so please close this if you want https://github.com/amilajack/eslint-plugin-compat/issues/206

```js
const x = null ?? "default string";

const adventurer = {
  name: "Alice",
  cat: {
    name: "Dinah",
  },
};

const dogName = adventurer.dog?.name;
```
`.eslintrc.json`
```json
{
  "extends": ["plugin:compat/recommended"],
  "env": {
    "browser": true,
    "es2020": true
  }
}
```
`package.json`
```json
  "browserslist": [
    "defaults"
  ],

  "devDependencies": {
    "eslint": "^7.3.1",
    "eslint-plugin-compat": "^3.7.0"
  }
```
It is falsely reporting that this modern syntax will work with the current default browsers.

**Comment by amilajack** [comment]
Are you transpiling with babel or ts?

**Comment by David-Else** [comment]
I have written an article that includes `eslint-plugin-compt` and used babel and SWC to compile in examples. I am not sure how your question relates to my issue, as this problem happens before any compilation. I am using the latest VS Code which has TS 3.9x language server and understands these modern features.

**Comment by amilajack** [comment]
If you're using a compiler then you shouldn't need to worry about the syntax not being supported. The plugin detects if you are using a transpiler and does not warn when using ESNext features if using a transpiler.

**Comment by David-Else** [comment]
I am not using a transpiler, I am using `eslint-plugin-compat` to lint for browser API feature compatibility and ESLint itself to lint for syntax/language features using `"esXXXX": true` as recommended in https://github.com/amilajack/eslint-plugin-compat/issues/206#issuecomment-504531256

My understanding is that this is the only current way to get things 100% working as `eslint-plugin-compat` does not lint for language features? (as my original issue shows). Cheers!
